src/riot-rs-runqueue/src/runqueue.rs

```diff
@@ -172,6 +191,31 @@ mod clist {
             }
         }
 
+        /// Removes a thread from the list.
+        ///
+        /// If the thread was the only thread in its runqueue, `Some` is returned
+        /// with the ID of the now empty runqueue.
+        pub fn del(&mut self, n: u8) -> Option<u8> {
```

Comment:
The `CList` submodule is internal and only uses u8 instead of `RunqueueId`/ `ThreadId` because it would otherwise cause issues with hax. See https://github.com/future-proof-iot/RIOT-rs/pull/283#issuecomment-2096316464.